PluginSkeletonGenerator/utils.py

```diff
@@ -0,0 +1,60 @@
+import os
+class FileUtils():
+    def __init__(self) -> None:
+        pass
+
+    @staticmethod
+    def replace_keywords(template, keywords):
+        code = template
+        data = keywords
+        for keyword, value in data.items():
+            code = code.replace(keyword,value)
+        return code
+    
+    @staticmethod
+    def read_file(template_name):
+        try:
+            with open(template_name, 'r') as template_file:
+                template = template_file.read()
+        except FileNotFoundError:
+            print(f"Could not load template: {template_name}")
+            return None
+        return template
+
+class Indenter():
+    def __init__(self) -> None:
+        self.indent_size = 0
+        self.indent = ""
+        self.indented_code = []
+
+    def indent_type(self, path):
+
+        extension = os.path.splitext(path)[1]
+        if extension == '.json':
+            self.indent = ' ' * 2
+        else:
+            self.indent = ' ' * 4
+
+    def process_indent(self, code, path):
+        self.indented_code.clear()
+        self.indent_type(path)
+        lines = code.splitlines()
+        self.indent_size = 0
+        for line in lines:
+            newline = line.strip()
+
+            if('~INDENT_RESET~' in newline):
+                self.indent_size = 0
+                continue
+            elif('~INDENT_INCREASE~' in newline):
+                self.indent_size +=1
+                continue
+            elif('~INDENT_DECREASE~' in newline):
+                self.indent_size -=1
+                continue
+            # rm\*n refers to remove line, therefore return this to delete line
+            elif('rm\*n' in newline):
+                continue
```

Comment:
do you think this is an appropriate approach to begin with, to have variables such as:
INDENT_RESET = '~INDENT_RESET~'
and then use your updated method 

In the templates I use these strings to indicate when an indent needs to be changed. I opted for these as I think its the simplest approach in general as its the most versatile. 
My first plan was to to use a "stack" and use {} brackets to accordingly calculate the indent which would work but not fully due to:
- namespaces causing extra indent (could obviously ignore the first/last two brackets but this wouldn't work on every file)
- not all files use brackets, CMake, ConfIn

So I decided to use "keywords" such as ~INDENT_INCREASE~ which are in the templates as it gives me most versatility, the main issue with it is that it can make the templates a bit uglier but if no one else meant to look at them, but imo it seemed like the easiest solution

Reply:
- I see and that makes sense. My first remark was due to the fact that right now you use magic strings. So, I thought simply storing them in variables and then using these variables makes it more scalable. However, since you have to update the templates anyway if we ever decide to update the keywords, and there is no way to use the variables there, I guess that's not a huge deal 😃
- But when it comes to the other changes, I simply thought that using a python dictionary, which is a hash table, instead of an `if-else` would both look cleaner and be more efficient. And if you decide to stick to magic string, you can simply put them inside this dictionary and the code should still work the same 😄

Reply:
```python
class Indenter():
    def __init__(self) -> None:
        self.indent_size = 0
        self.indent = ""
        self.indented_code = []
        self.indent_map = self.create_indent_map()

        self.INDENT_RESET = '~INDENT_RESET~'
        self.INDENT_INCREASE = '~INDENT_INCREASE~'
        self.INDENT_DECREASE = '~INDENT_DECREASE~'
        self.REMOVE_LINE = 'rm\*n'

    def create_indent_map(self):
        return {
            '.json': ' ' * 2,
            '.py' : ' ' * 4,
            '.h' : ' ' * 4,
            '.cpp' : ' ' * 4
        }

    def indent_type(self, path):
        extension = os.path.splitext(path)[1]
        self.indent = self.indent_map.get(extension, ' ' * 4) # Default case: 4 spaces

    def process_indent(self, code, path):
        if not code:
            return ''

        self.indented_code.clear()
        self.indent_type(path)
        lines = code.splitlines()
        self.indent_size = 0

        actions = {
            self.INDENT_RESET: lambda: setattr(self, 'indent_size', 0),
            self.INDENT_INCREASE: lambda: setattr(self, 'indent_size', self.indent_size + 1),
            self.INDENT_DECREASE: lambda: setattr(self, 'indent_size', self.indent_size - 1),
            self.REMOVE_LINE: lambda: None
        }

        for line in lines:
            newline = line.strip()

            action = actions.get(newline, None)
            if action:
                action()
                continue

            indented_line = self.indent * self.indent_size + newline
            self.indented_code.append(indented_line)
        return '\n'.join(self.indented_code)

```